### PR TITLE
[rllib] Remove deprecated policy optimizer package.

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

As a followup from previous discussion, remove the deprecated policy optimizers module. RST documentation for the execution plan API will be added in a follow-up PR.

### CODE REVIEW

**Comment by richardliaw** [comment]
wow nice; feel free to send over the replacement docs PR later

**Comment by michaelzhiluo** [comment]
It's finally happening! Rip Optimizers

File: rllib/utils/framework.py [hunk 1]
```diff
@@ -64,7 +64,10 @@ def try_import_tf(error=False):
     except AttributeError:
         tf1_module = tf_module
 
-    version = 2 if "2." in tf_module.__version__[:2] else 1
+    if not hasattr(tf_module, "__version__"):
```
Comment (sven1977) [inline rllib/utils/framework.py:67]: Thanks!

**Review by sven1977 — APPROVED** [review]
LGTM. Just one question: What's the reason we retired the custom make_workers option?

**Comment by ericl** [comment]
make_workers turned out to be only used as a hack for the legacy IMPALA. We could potentially add it back, though I'm not sure about the use cases that would need it.

**Pull request merged**